packages/react/modules/components/configure-adapter/configure-adapter.js

```diff
@@ -69,10 +69,20 @@ export default class ConfigureAdapter extends PureComponent<Props, State> {
     this.adapterState = nextAdapterState;
   };
   applyAdapterArgs = (nextAdapterArgs: AdapterArgs): void =>
-    this.setState(prevState => ({
-      ...prevState,
-      appliedAdapterArgs: nextAdapterArgs,
-    }));
+    this.setState(prevState => {
+      /**
+       * NOTE:
+       *   We can only unset `pendingAdapterArgs` after be actually perform
+       *   a batched `setState` otherwise outdated `adapterArgs` may be
+       *   applied as the `setState` is actually performed later.
+       */
+      this.pendingAdapterArgs = null;
+
+      return {
+        ...prevState,
+        appliedAdapterArgs: nextAdapterArgs,
+      };
+    });
```

Comment:
I don't fully get the comment... but I'm missing a ton of context anyways.

Would one of these two solutions be preferable?

```js
this.pendingAdapterArgs = null;
this.setState(prevState => {})
```

```js
this.setState(
  prevState => {},
  () => { this.pendingAdapterArgs = null; }
)
```

Just a suggestion as I've never seen a side-effect within the `setState`.

Reply:
Oh shit. Thanks. Will do